### DISCUSSION TITLE
Bad Request with Snowflake connection if you test the connection after its created?

### DISCUSSION THREAD

**Discussion by kuldeep773** [apache/airflow#21790] [General] [answered]
I am new to this forum. I have a weird scenario when testing the snowflake connection in airflow 2.2.4. 

The snowflake connection only test the connection successfully when it was created for the first time. Once you have saved and go back to the connection to test the Snowflake connection again, it return a "Bad Request" error. This is so confusing because the code works. Is this a bug with the Snowflake connector/hook?

If anyone of you also faced a similar issue, you expertise is greatly appreciated. There is a thread going on [here](https://stackoverflow.com/questions/70941912/im-getting-an-error-bad-request-while-establishing-an-connection-between-airf/71250997#71250997), that worth checking

**Comment by josh-fell** [comment, marked as answer]
Seems like slightly different things going on between retrying a successful connection vs. using the custom form fields in the UI to test the connection initially (from the StackOverflow post).

For the latter, there is a bug which prevents the test-connection functionality from using the custom form fields. However, this should be fixed with #21330.

For retrying a test on an existing Snowflake connection, do you just need to reenter the password or completely recreate the connection again? I believe you only need to reenter the password again IIRC.

**Reply by kuldeep773** [reply]
So far what I have observed is, for retrying I have to delete the connection . Simply providing password doesn't help :( .

**Reply by kuldeep773** [reply]
So far what I have observed is, for retrying I have to delete the connection . Simply providing password doesn't help :( .

**Reply by kuldeep773** [reply]
So far what I have observed is, for retrying I have to delete the connection . Simply providing password doesn't help :( .

**Reply by josh-fell** [reply]
Did some testing. If you delete all of the values from the custom form fields and reenter the password, you can test an existing connection without having to delete it entirely. Obviously this is a workaround.

I did confirm that this scenario doesn't occur on the `main` branch. All you need to do is reenter the password to retest the connection.

**Reply by kuldeep773** [reply]
Thank you so much for looking into this. Might be a stupid question but the main branch is on which version?  I can give it a try

**Reply by josh-fell** [reply]
`main` is technically development. It contains the latest-and-greatest which includes unreleased code (e.g. fixes like #21330). You can fork this repo and test in a [local dev environment](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#development-environments) (IMO [Breeze](https://github.com/apache/airflow/blob/main/BREEZE.rst) being the best approach).

**Reply by gigipillai** [reply]
Hello,
Kindly suggest the solution for this issue.
Get "Bad Request" while placing the Airflow Connectors with Snowflake and Azure Data Factory.

Regards,
Gigi

**Reply by kuldeep773** [reply]
> Hello, Kindly suggest the solution for this issue. Get "Bad Request" while placing the Airflow Connectors with Snowflake and Azure Data Factory.
> 
> Regards, Gigi

Try delete the connection and make a fresh one. The first time you test it, the connection would be successful. Save the connection and it will work. Unfortunately that was the workaround that worked for me

**Reply by gigipillai** [reply]
Thanks Kuldeep.
Tried the workaround as per https://stackoverflow.com/questions/70941912/im-getting-an-error-bad-request-while-establishing-an-connection-between-airf/71250997#71250997
But doesn't seem to work for me.
Do we need to also setup the Environment Variable or any other setting that should be taken care of?

Regards,
Gigi
